reconcile/utils/terrascript_aws_client.py

```diff
@@ -231,13 +233,37 @@ def __init__(self, integration: str, integration_prefix: str,
             ts += provider.template(
                 version="2.2.0"
             )
-
-            b = Backend("s3",
-                        access_key=config['aws_access_key_id'],
-                        secret_key=config['aws_secret_access_key'],
-                        bucket=config['bucket'],
-                        key=config['{}_key'.format(integration)],
-                        region=config['region'])
+            access_key_value = config['aws_access_key_id']
+            secret_key_value = config['aws_secret_access_key']
+            accounts = queries.get_aws_accounts(terraform_state=True)
```

Comment:
why query for the accounts again? filtered_accounts already holds the accounts queries in terraform_resources.py and you could query with `terraform_state=True` there

https://github.com/app-sre/qontract-reconcile/blob/826cc8683cba0dccbe9fae677d4c2f04d3c5828c/reconcile/terraform_resources.py#L467